daemons/controld/controld_join_dc.c

```diff
@@ -673,23 +673,168 @@ finalize_sync_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, voi
     }
 }
 
+/*!
+ * \internal
+ * \brief Free XML in a CIB client callback
+ *
+ * \param[in,out] xml  XML to free
+ */
 static void
-join_update_complete_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void *user_data)
+xml_callback_free_fn(void *xml)
 {
-    fsa_data_t *msg_data = NULL;
+    free_xml((xmlNode *) xml);
+}
```

Comment:
I do greatly prefer wrappers to casting function signatures. Might be worth formalizing in the style guide ...

Reply:
Casting a function is extremely convenient, bordering on necessary, for GLib and probably for some internal functions. The real issue is incompatible types. Technically a pointer-to-struct is not guaranteed to use the same representation as a pointer-to-void, pointer-to-char, etc.

I started writing up a doc guideline but it might be better to leave it alone. The compiler seems to warn if it's a problem.

If we do document it, we can probably make the guideline apply to struct and union types, and put it in the pointers section instead of the functions section.